src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,113 @@
+import cn from 'classnames';
+import { useContext, useState } from 'react';
+import { TodoItemType } from '../../types/TodoItemType';
+import { DispatchContext, TodosContext } from '../GlobalStateProvider';
+import { AllActions } from '../../types/Action';
+
+type Props = {
+  todo: TodoItemType,
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const todos = useContext(TodosContext);
+  const dispatch = useContext(DispatchContext);
+  const [isEditingMode, setIsEditingMode] = useState(false);
+  const [updatedName, setUpdatedName] = useState(todo.title);
+
+  const handleChange = (id: number) => {
+    const index = todos.findIndex(item => item.id === id);
+    const updated = todos[index];
+
+    updated.completed = !updated.completed;
+    dispatch({ type: AllActions.Update, payload: id, value: updated });
+  };
+
+  const handleDelete = () => {
+    dispatch({ type: AllActions.Remove, payload: todo.id });
+  };
+
+  const handleKeyDown
+    = (event: React.KeyboardEvent, updated: TodoItemType) => {
+      if (event.key === 'Enter' && updatedName.trim() !== '') {
```

Comment:
```suggestion
      if (event.key === 'Enter') {
        if (updatedName.trim()) {
          // do something
        } else {
          // do something else
        }
```
schematically  :arrow_up: 
it's easier to read than similar combined conditions